### PR TITLE
Added Support for Middleware Functions in `piral-fetch`

### PR DESCRIPTION
**Pull request by manuelroemer**
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes

### Description

Implements the proposed `piral-fetch` middleware changes from #645. See the issue for details.

### Remarks

_None._

### CODE REVIEW

**Review by FlorianRappl — COMMENTED** [review]
LGTM - we still need an entry in the `CHANGELOG.md` (this is for the new 1.4.1 - so a new entry is needed; first pull from `documentation` to have the most recent `develop` branch)

**Comment by manuelroemer** [comment]
Ok, will do. IIRC the next title in the changelog should then be `## 1.4.1 (tbd)`, right?

**Comment by FlorianRappl** [comment]
> Ok, will do. IIRC the next title in the changelog should then be `## 1.4.1 (tbd)`, right? Let me know if not. ;)

Yes that is right

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Piral Changelog
 
-## 1.4.0 (tbd)
+## 1.4.1 (tbd)
+
+- Added support for middleware functions in `piral-fetch` (#655)
```
Comment (FlorianRappl) [inline CHANGELOG.md:5]: You can use a reference to `#645` as #645 is the issue / story for it.

**Pull request merged**